src/lib.rs

```diff
@@ -1266,46 +1296,13 @@ impl<T: Counter> Histogram<T> {
     /// If the value is larger than the maximum representable value, it will be clamped to the
     /// max representable value.
     pub fn count_at(&self, value: u64) -> T {
-        // TODO panic on bad index
-        self[self.index_for_or_last(value)]
+        self.count_at_index(self.index_for_or_last(value)).expect("index is <= last_index()")
     }
 
     // ********************************************************************************************
     // Public helpers
     // ********************************************************************************************
 
-    /// Computes the matching histogram value for the given histogram bin.
-    ///
-    /// `index` must be no larger than `u32::max_value()`; no possible histogram uses that much
-    /// storage anyway.
-    pub fn value_for(&self, index: usize) -> u64 {
-        assert!((index as u64) < (u32::max_value() as u64), "index must be <= u32::max_value()");
-        // TODO what to do about indexes that are beyond what can be represented by this histogram?
```

Comment:
This TODO can go away since it's not possible when non-public (unless you're being lame in a test, which I am, to verify that it is broken as expected)